test/nn/test_load_state_dict.py

```diff
@@ -362,7 +365,162 @@ def test_load_state_dict_warn_assign(self):
             m.load_state_dict(state_dict)
 
 
+class MyLoadTensor(torch.Tensor):
+
+    # enable deepcopy
+    def new_empty(self, shape):
+        return type(self)(shape)
+
+    @classmethod
+    def __torch_function__(cls, func, types, args=(), kwargs=None):
+        kwargs = {} if kwargs is None else kwargs
+
+        def module_load_from(dest, src):
+            # always convert src to MyLoadTensor
+            if type(src) is torch.Tensor:
+                return MyLoadTensor(src)
+            elif type(src) is MyLoadTensor:
+                return src
+            elif type(src) is MyWrapperLoadTensor:
+                return MyLoadTensor(src._data)
+            else:
+                raise NotImplementedError(f"module_load_from(): unsupported type(src) {type(src)}")
+
+        def module_load_to(src, dest):
+            if type(dest) in {torch.nn.Parameter, torch.Tensor, MyLoadTensor, MyWrapperLoadTensor}:
+                return src
+            else:
+                raise NotImplementedError(f"module_load_to(): unsupported type(dest) {type(dest)}")
+
+        if func is torch.Tensor.module_load_from:
+            return module_load_from(*args, **kwargs)
+        elif func is torch.Tensor.module_load_to:
+            return module_load_to(*args, **kwargs)
+        elif func is torch.Tensor.detach:
+            return MyLoadTensor(args[0].data.detach())
+        else:
+            with torch._C.DisableTorchFunctionSubclass():
+                return func(*args, **kwargs)
+
+
+class MyWrapperLoadTensor(torch.Tensor):
```

Comment:
Inherit from MyLoadTensor to reduce duplication?

Reply:
Hmmm one is a wrapper subclass and one is not though, can I still inherit from the non-wrapper MyLoadTensor